.pipelines/cni/singletenancy/linux-cniv2-template.yaml

```diff
@@ -17,25 +17,48 @@ parameters:
 # + CNS restart and validates the state
 
 stages:
-  - stage: create_${{ parameters.name }}
-    variables:
-      commitID: $[ stagedependencies.setup.env.outputs['SetEnvVars.commitID'] ]
-    pool:
-      name: $(BUILD_POOL_NAME_DEFAULT)
-    dependsOn:
-      - setup
-    displayName: "Create Cluster - ${{ parameters.clusterName }}"
-    jobs:
-      - job: create_aks_cluster_with_${{ parameters.name }}
-        steps:
-          - template: ../load-test-templates/create-cluster-template.yaml
-            parameters:
-              clusterType: ${{ parameters.clusterType }}
-              clusterName: ${{ parameters.clusterName }}-$(commitID)
-              nodeCount: ${{ parameters.nodeCount }}
-              vmSize: ${{ parameters.vmSize }}
-              region: $(LOCATION)
-              osSKU: ${{ parameters.osSKU }}
+  - ${{ if eq(parameters.os, 'linux') }}:
+    - stage: create_${{ parameters.name }}
+      condition: and( succeeded(), or( and( contains(variables.CONTROL_CNI, 'cniv2'), contains(variables.CONTROL_OS, 'linux') ) , contains(variables.CONTROL_CNI, 'all') ) )
```

Comment:
In case of `CONTROL_CNI` as all and CONTROL_OS as `windows` the above state is true ? Where as my intentions is to test all CNI for windows. 
Or the line 20 won't be triggered somehow ? Trying to understand this.

Reply:
Yes, it would be true due to me missing an interaction/check that I want down below. It will be fixed.

Line 20 intent is to control expected comparisons based on the template call (subsequently the same for line 41) so that we can match up the `CONTROL_OS` calls with `parameter.os`. This is because we cannot pass in `parameter.xxx` because they are compile time variables versus the runtime variables of `variable.xxx`. Ideally this would all be on one line and one stage block, but c'est la vie. 

As it is now, if `CONTROL_CNI = all` then all clusters will be created, which was not intended. Correct logic would be this 
`and( succeeded(),`
`and( `
`   or( contains(variables.CONTROL_CNI, 'cniv2'), contains(variables.CONTROL_CNI, 'all') ),`  
`   or( contains(variables.CONTROL_OS, 'linux'), contains(variables.CONTROL_OS, 'all') )`
`)`

Reply:
Expected variations are listed above in the PR description.